var/spack/repos/builtin/packages/hdf5-vfd-gds/package.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+class Hdf5VfdGds(CMakePackage, CudaPackage):
+    """This pacakge enables GPU Direct Storage Virtual File Driver in HDF5."""
+
+    # Package info
+    homepage    = 'https://github.com/hpc-io/vfd-gds'
+    url = 'https://github.com/hpc-io/vfd-gds/archive/refs/tags/1.0.1.tar.gz'
+    git         = 'https://github.com/hpc-io/vfd-gds.git'
+    maintainers = ['hyoklee', 'lrknox']
+
+    # Versions
+    version('master', branch='master')
+    version('1.0.1', sha256='00e125fd149561be991f41e883824de826d8add604aebccf103a4fb82d5faac2', default=True)
+    version('1.0.0', sha256='6b16105c7c49f13fc05784ee69b78d45fb159270c78d760689f9cd21e230ddd2')
+    
+    # Dependencies
+    depends_on('cuda')    
```

Comment:
A `CudaPackage` has already a conditional dependency on `cuda` and a `cuda` boolean variant, hence adding an unconditional dependency on `cuda` seems wrong. I think what might work here is:
```
conflicts('~cuda')
```
if the aim is to have cuda always enabled.